lhotse/recipes/timit.py

```diff
@@ -0,0 +1,139 @@
+
+    tar_name = f'timit.zip'
+    tar_path = target_dir / tar_name
+    if force_download or not tar_path.is_file():
+        urlretrieve_progress(f'{base_url}', filename=tar_path, desc=f'Downloading {tar_name}')
+
+
+def prepare_timit(
+        corpus_dir: Pathlike,
+        splits_dir: Pathlike,
+        output_dir: Optional[Pathlike] = None,
+        num_jobs: int = 1
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+    """
+    """
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f'No such directory: {corpus_dir}'
+
+    splits_dir = Path(splits_dir)
+    assert corpus_dir.is_dir(), f'No such directory: {splits_dir}'
+
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    manifests = defaultdict(dict)
+    dataset_parts = ['TRAIN', 'DEV', 'TEST']
+
+    punctuation_strings = string.punctuation
+
+    with ThreadPoolExecutor(num_jobs) as ex:
+        for part in dataset_parts:
+          wav_files = []
+          file_name = ''
+
+          if part == 'TRAIN':
+            file_name = os.path.join(splits_dir, 'train_samples.txt') 
+          elif part == 'DEV':
+            file_name = os.path.join(splits_dir, 'dev_samples.txt')
+          else:
+            file_name = os.path.join(splits_dir, 'tst_samples.txt')
+          wav_files = []
+          with open(file_name, 'r') as f:
+            lines = f.readlines()
+            for line in lines:
+                items = line.strip().split(' ')
+                wav = os.path.join(corpus_dir, items[-1])
+                wav_files.append(wav)
+            print(f'{part} dataset manifest generation.')
+            recordings = []
+            supervisions = []
+
+            for wav_file in tqdm(wav_files):
+                items = wav_file.split('/')
+                idx = items[-2] + '-' + items[-1][:-4]
+                speaker = items[-2]
+                transcript_file = wav_file[:-3] + 'PHN' ###the phone file
+                if not Path(wav_file).is_file():
+                    logging.warning(f'No such file: {wav_file}')
+                    continue
+                if not Path(transcript_file).is_file():
+                    logging.warning(f'No transcript: {transcript_file}')
+                    continue
+                text = []
+                with open(transcript_file, 'r') as f:
+                    lines = f.readlines()
+                    for line in lines:
+                        phone = line.rstrip('\n').split(' ')[-1]
+                        text.append(phone)
+                    text = ' '.join(text).replace('h#', 'sil')
+
+                for i in punctuation_strings:
+                    if i != "'":
+                        text = text.replace(i, '')
+
+                recording = Recording.from_file(path=wav_file, recording_id=idx)
+                recordings.append(recording)
+                segment = SupervisionSegment(
```

Comment:
Are supervisions in this recipe sentences, words, or individual phones? If they are larger than phones, is it possible to add phone alignments in the SupervisionSegment object? I think TIMIT should have that data.